Apsitvarkom.Models/Location.cs

```diff
@@ -1,36 +1,24 @@
-﻿namespace Apsitvarkom.Models;
+﻿using FluentValidation;
+
+namespace Apsitvarkom.Models;
 
 /// <summary>
 /// Class for storing locations.
 /// </summary>
 public class Location
 {
-    private double _latitude;
-    private double _longitude;
-
     /// <summary>Latitude of the location. Cannot be less than -90.0 degrees or exceed 90.0 degrees.</summary>
-    public double Latitude
-    {
-        get => _latitude;
-        set
-        {
-            if (value is >= -90.0 and <= 90.0)
-                _latitude = value;
-            else
-                throw new ArgumentOutOfRangeException(nameof(Latitude), $"Latitude cannot be less than -90.0 or exceed 90.0 degrees, but was {value}.");
-        }
-    }
+    public double Latitude { get; set; }
     
     /// <summary>Longitude of the location. Cannot be less than -180 degrees or exceed 180 degrees.</summary>
-    public double Longitude
+    public double Longitude { get; set; }
+}
+
+public class LocationValidator : AbstractValidator<Location>
+{
+    public LocationValidator()
     {
-        get => _longitude;
-        set
-        {
-            if (value is >= -180.0 and <= 180.0)
-                _longitude = value;
-            else
-                throw new ArgumentOutOfRangeException(nameof(Longitude), $"Longitude cannot be less than -180.0 or exceed 180.0 degrees, but was {value}.");
-        }
+        RuleFor(data => data.Longitude).GreaterThanOrEqualTo(-180.0).LessThanOrEqualTo(180.0);
```

Comment:
I would consider using [InclusiveBetween](https://docs.fluentvalidation.net/en/latest/built-in-validators.html#inclusivebetween-validator), as I believe we would achieve the same with a more informative error message (error messages examples are in the linked docs).

Reply:
You are right here